iree/turbine/kernel/wave/generate_bounds_exprs.py

```diff
@@ -76,4 +77,16 @@ def is_read_write(node: fx.Node):
         if not bounds:
             continue
 
+        # Build map for all scaled dims to it's scaled expression.
+        scaled_dim_map = {
+            infer_dim(expr): expr
+            for expr in node.type.symbolic_shape
+            if is_scaled_dim(expr)
+        }
+        # Apply scaled expression to scaled dim if any.
+        bounds = {
```

Comment:
I wonder if we should handle it in `find_index_bounds` https://github.com/iree-org/wave/blob/main/iree/turbine/kernel/wave/utils/general_utils.py#L179 instead.

Reply:
I was debating between the two functions, done (I think) lmk if you still do not think it's in the right place :)